### PR TITLE
Auto-retry discord exceptions in message regen

### PR DESCRIPTION
**Pull request by evandixon**

### CODE REVIEW

**Comment by End45** [comment]
If we want to merge this, I think this logic should probably be moved to `process_update_hack_list`.

**Comment by evandixon** [comment]
> If we want to merge this, I think this logic should probably be moved to `process_update_hack_list`.

I disagree. The responsibility of this file is interacting with the Discord API, and the responsibility of process_update_hack_list is to iterate through hacks and be a bot command. Ideally the discord module would handle throttling for us, but here we are. If we were to further refactor, we'd separate this class from discord interaction entirely and use a helper class of our own.

**Comment by End45** [comment]
Hm, you're right. Discord logic doesn't belong in `hacks_mgmnt`. But it also feels wrong that the method that takes care of updating a single message has to account for rate limits that can happen when it's called multiple times. There should be some centralized way of waiting before retrying any operation that interacts with Discord's API.

But since that would take a significant amount of effort and it's unlikely that we will have to deal with something like this, I think this solution works for now. We can leave a TODO similar to the one I put at https://github.com/SkyTemple/swablu/blob/ed34ed85e09a1e3dde2160746955e2ab2dfb083b/swablu/specific/hacks_mgmnt.py#L129.

Bt the way, we probably need to get rid of that delay in `process_update_hack_list` if we want to merge this.

File: swablu/discord_util.py [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
+            else:
+                message = await channel.fetch_message(message_id)
+                if message.content != text:
+                    await message.edit(content=text)
+            break
+        except Exception:
+            try_count += 1
+            time.sleep(try_count * 5)
+            if try_count >= 10:
+                raise
```
Comment (End45) [inline swablu/discord_util.py:28]: Shouldn't `if try_count >= 10: raise` be placed before the sleep() call?

**Review by End45 — APPROVED** [review]
Looks good enough.

**Pull request merged**